sql/core/src/test/scala/org/apache/spark/sql/DatasetPrimitiveSuite.scala

```diff
@@ -28,6 +28,8 @@ case class SeqClass(s: Seq[Int])
 
 case class ListClass(l: List[Int])
 
+case class JListClass(l: java.util.List[Int])
```

Comment:
Alternatively you can improve `JavaTypeInference` to support specific java list and test it with java bean in `JavaDatasetSuite`